src/rtc.c

```diff
@@ -758,6 +807,36 @@ void RTC_StartAlarm(alarm_t name, uint8_t day, uint8_t hours, uint8_t minutes, u
     UNUSED(mask);
 #endif /* !STM32F1xx */
 
+    /* Set RTC_Alarm */
+    HAL_RTC_SetAlarm_IT(&RtcHandle, &RTC_AlarmStructure, RTC_FORMAT_BIN);
+    HAL_NVIC_SetPriority(RTC_Alarm_IRQn, RTC_IRQ_PRIO, RTC_IRQ_SUBPRIO);
+    HAL_NVIC_EnableIRQ(RTC_Alarm_IRQn);
+  } else {
+    /* We have an SubSecond alarm to set in RTC_BINARY_MIX mode */
```

Comment:
Seems nothing indicate we are in binary mode, only that arguments is not correct to start calendar alarm.

Reply:
+ checking the RTC mode is binary with 
`else if (RtcHandle.Init.BinMode != MODE_BINARY_NONE) {`